nvflare/apis/impl/bcast_manager.py

```diff
@@ -53,6 +54,13 @@ def check_task_exit(self, task: Task) -> Tuple[bool, TaskCompletionStatus]:
             # nothing has been sent - continue to wait
             return False, TaskCompletionStatus.IGNORED
 
+        if task.targets:
+            num_expected_clients = len(task.targets)
+        else:
+            engine = task.get_prop("___engine")
+            assert isinstance(engine, ServerEngineSpec)
+            num_expected_clients = len(engine.get_clients())
+
```

Comment:
instead of doing these checks in `check_task_exit`,
What about we move these lines in the `__init__` function?